src/ros_filter.cpp

```diff
@@ -2015,6 +2091,19 @@ namespace RobotLocalization
       accelPub_.publish(filteredAcceleration);
     }
 
+    // Publish rejected measurement topics since last predict cycle, if desired
+    if (publishRejectedMeasurements_)
+    {
+      std::vector<std::string> rejectedMsgs;
+      filter_.getRejectedMeasurementTopics(rejectedMsgs);
+      for(auto msg_topic : rejectedMsgs)
+      {
+        std_msgs::String sendMsg;
+        sendMsg.data = msg_topic;
+        rejectedMeasurementsPub_.publish(sendMsg);
+      }
+    }
+
```

Comment:
Is the filter intended to get cleared after each predict cycle? If so, where does that happen? since I don't see it explicitly given here...

Reply:
See other comments. The vector is cleared in `filter_.getRejectedMeasurementTopics(rejectedMsgs)`